crates/router/src/connector/datatrans/transformers.rs

```diff
@@ -445,18 +473,20 @@ impl TryFrom<types::PaymentsCaptureResponseRouterData<DataTransCaptureResponse>>
         item: types::PaymentsCaptureResponseRouterData<DataTransCaptureResponse>,
     ) -> Result<Self, Self::Error> {
         let status = match item.response {
-            DataTransCaptureResponse::Error(error) => match error
-                .message
-                .unwrap_or(consts::NO_ERROR_MESSAGE.to_string())
-                .as_str()
-            {
-                "already settled" => common_enums::AttemptStatus::Charged,
-                _ => common_enums::AttemptStatus::Failure,
-            },
-            DataTransCaptureResponse::Empty => match item.http_code {
-                204 => common_enums::AttemptStatus::Charged,
-                _ => common_enums::AttemptStatus::Failure,
-            },
+            DataTransCaptureResponse::Error(error) => {
+                if error.message == *"already settled" {
```

Comment:
```suggestion
                if error.message == "already settled".to_string() {
```

Reply:
warning: this creates an owned instance just for comparison
   --> crates/router/src/connector/datatrans/transformers.rs:517:37
    |
517 |                 if error.message == "already settled".to_string() {
    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `*"already settled"`
    
    
    Was getting this warning while running the 
    cargo clippy --all-features --all-targets